src/Core/Auth/Repositories/Cosmos/GrantRepository.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Net;
+using System.Text.Json.Serialization;
+using System.Text.Json;
+using Bit.Core.Auth.Models.Data;
+using Bit.Core.Settings;
+using Microsoft.Azure.Cosmos;
+
+namespace Bit.Core.Auth.Repositories.Cosmos;
+
+public class GrantRepository : IGrantRepository
+{
+    private readonly CosmosClient _client;
+    private readonly Database _database;
+    private readonly Container _container;
+
+    public GrantRepository(GlobalSettings globalSettings)
+        : this(globalSettings.IdentityServer.CosmosConnectionString)
+    { }
+
+    public GrantRepository(string cosmosConnectionString)
+    {
+        var options = new CosmosClientOptions
+        {
+            Serializer = new SystemTextJsonCosmosSerializer(new JsonSerializerOptions
+            {
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+                WriteIndented = false,
+            })
+        };
+        _client = new CosmosClient(cosmosConnectionString, options);
+        _database = _client.GetDatabase("identity");
+        _container = _database.GetContainer("grant");
```

Comment:
We could, but I am not sure why we would name them something different. The connection string for the Cosmos account to use is config driven already. Is there a use case you can think of that would benefit from this coming from a config prop?

Not sure I understand the property names point you made.

Reply:
I'm probably just overthinking things for a change here via config we would likely never do, like a need to move to multiple containers.

My other comment is about how proprietary we make the name of a configuration property e.g. "Cosmos" when maybe we could leave it more ambiguous for the future should:

- it not be called Cosmos anymore
- we support other document or table database types behind this
- something else.

This is more of an OCD thing of mine for future-proofing and again it could just not happen or we'd move to a whole new configuration naming.

Reply:
I think this is overthinking it a bit. This implementation is pretty specific to cosmos and trying to genericize it would make it less clear what is going on here, IMO.